File: QuickBuy.Dominio/Entidades/Pedido.cs
using System;
using System.Collections.Generic;
using System.Linq;
using QuickBuy.Dominio.OBjetoDeValor;

namespace QuickBuy.Dominio.Entidades
{
    public class Pedido : Entidade
    {
        public int Id { get; set; }
        public DateTime DataPedido { get; set; }
        public int UsuarioId { get; set; }
        public ICollection<ItemPedido> ItensPedidos { get; set; }
        public DateTime DataPrevisaoEntrega { get; set; }
        public string CEP { get; set; }
        public string Estado { get; set; }
        public string Cidade { get; set; }
        public string EderecoCompleto { get; set; }
        public int NumeroEndereco { get; set; }
        public int FormaPagamentoId { get; set; }
        public FormaPagamento FormaPagamento { get; set; }

        public override void Validate()
        {
            LimparMensagensValidacao();
            if (!ItensPedidos.Any())
                AdicionarCritica("Critica : pedido nao pode ficar sem item de pedido");

            if (string.IsNullOrEmpty(CEP))
                AdicionarCritica("Critica : CEP deve estar preenchido");

            if(FormaPagamentoId == 0)
                AdicionarCritica("Critica : Nao foi infomado a forma de pagamento");

            
        }
    }
}

File: QuickBuy.Web/ClientApp/src/app/modelo/pedido/pedido.ts
import { ItemPedido } from "../item-pedido/itemPedido";

export class Pedido {
  public id: number;
  public dataPedido: Date;
  public usuarioId: number;
  public dataPrevisaoEntrega: Date;
  public cep: string;
  public estado: string;
  public cidade: string;
  public enderecoCompleto: string;
  public numeroEndereco: number;
  public formaPagamentoId: number;
  public itensPedido: ItemPedido[];

  constructor() {
    this.itensPedido = [];
  }
}
